### DISCUSSION TITLE
Nominal typing on ID types?

### DISCUSSION THREAD

**Discussion by olalonde** [prisma/prisma#21009] [Q&A] [answered]
### Question

I was wondering if there has been any previous discussion on introducing strongly typed ID types (e.g. using this "hack" `type UserId = string & { readonly __brand: unique symbol }`)? Since ID from one table are usually not interchangeable with IDs from another table, it would be a useful feature. I have been using this hack in one of my internal libraries and it really helps catch bugs and facilitates refactoring.

**Comment by ludralph** [comment, marked as answer]
Hi @olalonde 👋 

Thank you for raising this question.

I have yet to come across any discussion for introducing strongly typed ID types. Can you please open a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) documenting how that would look like so that our engineers can take a look at it?

**Reply by olalonde** [reply]
Sure, I created this issue:

https://github.com/prisma/prisma/issues/21024
